packages/firebase/useAuth/index.ts

```diff
@@ -0,0 +1,22 @@
+import { computed, ComputedRef, ref, Ref } from 'vue'
+import firebase from 'firebase'
+import 'firebase/auth'
+
+export interface FirebaseAuthOptions {
+  isAuthenticated: ComputedRef<boolean>
+  user: Ref<firebase.User | null>
+}
+
+export function useAuth() {
+  const { auth } = firebase
+
+  const user = ref<firebase.User | null>(auth().currentUser)
```

Comment:
Is calling is sync, it looks like there is not different to call it at the beginning of the setup(). As long as you do the initialization in main.js I thought it should be fine. What's the problem you are running into?